File: Code/src/main/java/com/jmd/ApplicationSetting.java
package com.jmd;

import java.io.File;
import java.util.ArrayList;

import com.jmd.common.Setting;
import com.jmd.util.CommonUtils;

import lombok.Getter;

public class ApplicationSetting {

    private static final File path = new File(System.getProperty("user.dir") + "/setting");
    private static final File file = new File(System.getProperty("user.dir") + "/setting/setting");

    @Getter
    private static Setting setting;

    static {
        if (!path.exists() && !path.isFile()) {
            path.mkdir();
        }
        if (!file.exists() && !file.isFile()) {
            setting = createDefault();
        } else {
            setting = loadSettingFile();
        }
    }

    public static void save() {
        try {
            CommonUtils.saveObj2File(setting, file.getAbsolutePath());
        } catch (Exception e) {
            e.printStackTrace();
        }
    }

    public static void save(Setting _setting) {
        setting = _setting;
        try {
            CommonUtils.saveObj2File(_setting, file.getAbsolutePath());
        } catch (Exception e) {
            e.printStackTrace();
        }
    }

    private static Setting createDefault() {
        var _setting = new Setting();
        _setting.setThemeType(3);
        _setting.setThemeName("Flatlaf IntelliJ");
        _setting.setThemeClazz("com.formdev.flatlaf.FlatIntelliJLaf");
        _setting.setAddedLayers(new ArrayList<>());
        try {
            CommonUtils.saveObj2File(_setting, file.getAbsolutePath());
        } catch (Exception e) {
            e.printStackTrace();
        }
        return _setting;
    }

    private static Setting loadSettingFile() {
        var _setting = new Setting();
        try {
            _setting = (Setting) CommonUtils.readFile2Obj(file);
        } catch (Exception e) {
            e.printStackTrace();
        }
        return _setting;
    }

}

File: Code/src/main/java/com/jmd/ApplicationSetting.kt
package com.jmd

import com.jmd.common.Setting
import com.jmd.util.MyFileUtils
import java.io.File

object ApplicationSetting {

    private val path = File(System.getProperty("user.dir") + "/setting")
    private val file = File(System.getProperty("user.dir") + "/setting/ApplicationSetting")

    private var setting: Setting

    init {
        if (!path.exists() && !path.isFile) {
            path.mkdir()
        }
        if (!file.exists() && !file.isFile) {
            setting = createDefault()
        } else {
            setting = loadSettingFile()
        }
    }

    @JvmStatic
    fun getSetting(): Setting {
        return setting
    }

    @JvmStatic
    fun save() {
        try {
            MyFileUtils.saveObj2File(setting, file.absolutePath)
        } catch (e: Exception) {
            e.printStackTrace()
        }
    }

    @JvmStatic
    fun save(s: Setting) {
        setting = s
        try {
            MyFileUtils.saveObj2File(setting, file.absolutePath)
        } catch (e: Exception) {
            e.printStackTrace()
        }
    }

    private fun createDefault(): Setting {
        val s = Setting("default")
        try {
            MyFileUtils.saveObj2File(s, file.absolutePath)
        } catch (e: Exception) {
            e.printStackTrace()
        }
        return s
    }

    private fun loadSettingFile(): Setting {
        var s: Setting
        try {
            s = MyFileUtils.readFile2Obj(file) as Setting
        } catch (e: Exception) {
            s = createDefault()
            e.printStackTrace()
        }
        return s
    }

}